.github/workflows/build-local.yml

```diff
@@ -0,0 +1,76 @@
+# Build directly on the GitHub runner
+name: build-local
+on:
+  workflow_dispatch:
+  workflow_call:
+
+concurrency:
+  group: build-${{github.ref}}-${{github.event.pull_request.number || github.run_number}}-${{github.workflow}}
+  cancel-in-progress: true
+
+jobs:
+  ubuntu-latest:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        compiler: ['gcc-10', 'gcc-12', 'clang-11', 'clang-15']
+    env:
+      CCACHE_DIR: "${{github.workspace}}/.ccache"
+      CCACHE_MAXSIZE: "10G"
+    steps:
+      - name: Install dependencies
+        run: |
+          sudo apt-get -y update
+          sudo apt-get -y install \
+            ccache cmake ninja-build python3 \
+            nlohmann-json3-dev libgtest-dev libhepmc3-dev \
+            ${{matrix.compiler}}
+      - name: Check out Celeritas
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 255
+          fetch-tags: true
```

Comment:
Do we need the tag and the 255 depth for the quick build?  Wouldn't the already downloaded by github repo be sufficient?